third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java

```diff
@@ -109,8 +109,8 @@ private Flux<MessageResult> getMailboxMessagesOfAUser(Username username, Mailbox
         return Mono.from(mailboxManager.getMailboxReactive(mailboxMetaData.getId(), mailboxSession))
             .map(Throwing.function(MessageManager::getMailboxEntity))
             .flatMapMany(Throwing.function(mailbox -> mapperFactory.getMessageMapper(mailboxSession).findInMailboxReactive(mailbox, MessageRange.all(), MessageMapper.FetchType.METADATA, UNLIMITED)))
-            .doOnNext(mailboxMessageMetaData -> context.incrementHamMessageCount())
             .filter(mailboxMessageMetaData -> afterDate.map(date -> mailboxMessageMetaData.getInternalDate().after(date)).orElse(true))
```

Comment:
Q:
why we don't check null for case `mailboxMessageMetaData.getInternalDate()` ?

Reply:
As I understand, we always write not null `internalDate` to our DB. This is only null when migration mistake.

Reply:
BTW I don't think we are accepting creating an object with null internal date (throw NPE like this) but predicate checking instead (throwing some IllegalArgumentException). Anyway, let's say RuntimeException generally and these tests are mainly proving what happened with a null internal date in the pre-prod Cassandra table.